assembl/static2/js/app/components/cookies/cookiesSelector.jsx

```diff
@@ -0,0 +1,92 @@
+// @flow
+import React from 'react';
+import { Translate } from 'react-redux-i18n';
+import { Button } from 'react-bootstrap';
+import classnames from 'classnames';
+import CookieToggle from './cookieToggle';
+import type { CookieObject } from './cookieToggle';
+
+type CookiesSelectorProps = {};
+
+type CookiesSelectorState = {
+  activeKey: ?string,
+  show: boolean,
+  cookies: Array<CookieObject>
+}
+
+
+class CookiesSelector extends React.Component<CookiesSelectorProps, CookiesSelectorState> {
+  constructor(props: CookiesSelectorProps) {
+    super(props);
+    const cookiesList = document.cookie.split(' ');
+    const cookies = cookiesList.map(cookie => ({ ...this.getCookieObject(cookie), accepted: true }));
+    this.state = {
+      activeKey: 'essential',
+      show: true,
+      cookies: cookies
+    };
+  }
+
+  getCookieObject = (cookie: string) => {
+    if (cookie.startsWith('assembl_session')) {
+      return { category: 'essential', name: 'userSession' };
+    }
+    if (cookie.startsWith('_LOCALE_')) {
+      return { cateogry: 'essential', name: 'locale' };
+    }
+    if (cookie.startsWith('_pk_')) {
+      return { category: 'analytics', name: 'piwik' };
+    }
+    return { category: 'other', name: cookie };
+  };
+
+  handleToggle = (updatedCookie: CookieObject) => {
+    const { cookies } = this.state;
+    const filteredCookies = cookies.map(c => (c.name === updatedCookie.name ? updatedCookie : c));
+    this.setState({ cookies: filteredCookies });
+  }
+
+  saveChanges = () => {
+    // saves the new cookies settings stored in localstate in the backend
```

Comment:
for piwik, you need to enable the optout module and include the related code in the page via an iframe.

Reply:
I'm not sure it's possible via iframe, but a link to the optout page is also good.

Reply:
Nah, iframe should work. It won't look good, but it's possible to offer the iframe inside of a modal, which is what V1 did too.